README.md

```diff
@@ -28,6 +32,121 @@ Project is written in TypeScript and provides type safety out of the box. No Flo
 -   [Server Side Rendering with `useStaticRendering`](#server-side-rendering-with-usestaticrendering)
 -   [Why no Provider/inject?](#why-no-providerinject)
 
+## Overview
+
+This library has two goals:
+
+1. Make it possible to introduce reactivity to React function components
+2. Make it possible to have local observable state in React function components
+
+This library does _not_ support class based components. If you need class based component support, use [mobx-react](https://github.com/mobxjs/mobx-react), which wraps this library.
+
+### Adding reactivity to React components
+
+There are three ways to add reactivity to components: `observer(fn)`, `<Observer>{renderFn}</Observer>` and `useObserver(renderFn)`.
+In most cases, there difference doesn't matter that much, but here is an overview of the three different approaches:
+
+```javascript
+import { observable } from "mobx"
+import { Observer, useObserver, observer }  from "mobx-react-lite"
+
+const person = observable({
+    name: "John"
+})
+
+const P1 = observer(function P1({ person }) {
+    return <h1>{person.name}</h1>
+})
+
+const P2 = ({ person }) => (
+    <Observer>{() => <h1>{person.name}</h1>}</Observer>
+)
+
+const P3 = ({ person }) => {
+    return useObserver(() => (
+        <h1>{person.name}</h1>
+    ))
+})
+
+ReactDOM.render(<div>
+    <P1 person={person} />
+    <P2 person={person} />
+    <P3 person={person} />
+</div>)
+
+setTimeout(() => {
+    person.name = "Jane"
+}, 1000)
+```
+
+When using `observer`, the component `props` will be made automatically observable (so that they can be used in reactions and computed values), and `React.memo` will be applied automatically to the comment.
+
+`useObserver` keeps the component tree flat as it doesn't introduce a higher order component, but to make props observable, the [`useAsObservableSource`](#useasobservablesourcetstate-t-t) hook should be used.
+
+`<Observer>` is slightly better optimized compared to `useObserver`, but does introduce an additional component in the component tree.
```

Comment:
In case you use it on some enclosed part of the render tree, then yea, it's probably faster, but wrapping a whole component body into it is the same as `useObserver`, isn't?

Reply:
I was referring to this: as soon as `useAsObservableSource` is used (or `useLocalStore` with the api change in #142), `useObserver` introduces double re-renderings: https://github.com/mobxjs/mobx-react-lite/issues/94#issuecomment-484085373

Reply:
Yea, so it's in specific scenarios. They way you wrote it feels like it's always faster and people should be favoring that. It's slightly misleading.